pkg/cnb/buildpack_retriever.go

```diff
@@ -0,0 +1,112 @@
+package cnb
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/v1"
+
+	"github.com/pivotal/kpack/pkg/apis/experimental/v1alpha1"
+)
+
+type BuildpackClient interface {
+	Fetch(keychain authn.Keychain, repoName string) (v1.Image, error)
+}
+
+type BuildpackRetriever struct {
+	Keychain authn.Keychain
+	Client   BuildpackClient
+	Store    *v1alpha1.Store
+}
+
+type BuildpackNotFound error
```

Comment:
Why did we create this type?

Reply:
It seemed nice to define a domain specific error even if we never end up using it to trigger code paths.